src/hooks/use-cart.ts

```diff
@@ -1,17 +1,43 @@
+import { useState } from 'react';
 import {
     useCartTotals,
     useCartData,
     useUpdateCartItemQuantity,
     useRemoveItemFromCart,
+    useAddToCart,
 } from '~/api/api-hooks';
 import { useEcomAPI } from '~/api/ecom-api-context-provider';
 
 export const useCart = () => {
     const ecomAPI = useEcomAPI();
+    const [updatingCartItems, setUpdatingCartItems] = useState<string[]>([]);
+
     const { data: cartData } = useCartData();
-    const { data: cartTotals } = useCartTotals();
-    const { trigger: updateItemQuantity } = useUpdateCartItemQuantity();
-    const { trigger: removeItem } = useRemoveItemFromCart();
+    const { data: cartTotals, isValidating: isCartTotalsValidating } = useCartTotals();
+
+    const { trigger: triggerUpdateItemQuantity } = useUpdateCartItemQuantity();
+    const { trigger: triggerRemoveItem } = useRemoveItemFromCart();
+    const { trigger: triggerAddToCart, isMutating: isAddingToCart } = useAddToCart();
+
+    const updateItemQuantity = ({ id, quantity }: { id: string; quantity: number }) => {
+        setUpdatingCartItems((prev) => [...prev, id]);
+        triggerUpdateItemQuantity(
```

Comment:
And even then, I don't know. I can send two requests at a small interval to update quantity of item A, once the first request completes, it will remove the item from the list.

If it's only used for the loader, then it's ok.

Reply:
Oh, it's quite hard to make such two requests. We have debounce for fast quantity changes (300 ms), after that request is sent and cart item is disabled under the loader. And loader disappears because the item is removed from an array on request success or error.

As for failure, let me do it in separate PR:

<img width="375" alt="Screenshot 2024-10-15 at 19 10 53" src="https://github.com/user-attachments/assets/e3f77c57-70e5-439a-9dbe-e53eb3e519bc">

<img width="1066" alt="Screenshot 2024-10-15 at 19 10 33" src="https://github.com/user-attachments/assets/83b99025-fb4e-4237-820d-d7bef08e429a">